src/js/src/store/modules/ngram.store.js

```diff
@@ -0,0 +1,116 @@
+import { requestService } from '../../services/RequestService'
+
+
+// Global ngram state.
+const initialState = () => ({
+  query: '',
+  // cache for current query in case server rejects it
+  attemptedQuery:'',
+  results: {},
```

Comment:
state.results isn't really used for anything significant. You put the current result in it after a successfull search, but use the state.datasets for drawing the chart. The only real use, as far as i can see, is in SearchBox.vue where you use it to check if the "reset"-button should be shown - but you can just look at the state.dataset variable instead. I think it should be removed to nice up the code :-)

Reply:
Another good catch!